### ISSUE TITLE
Serializer field with 'source' mapping to nullable foreign key field results in error while serializing

### ISSUE THREAD

**Issue by ozgurakcali** [encode/django-rest-framework#5750]
## Steps to reproduce
Create a model with a nullable foreign key field

```
class MyModel(models.Model):
    foreign_field = models.ForeignKey(OtherModel, null=True)

class MyModelSerializer(serializers.ModelSerializer):
    foreign_field_attribute = serializers.CharField(source='foreign_field.my_attribute', allow_null=True, required=False)
```

## Expected behavior
While retrieveing an entry with 'foreign_field' set to null, object should be serialized with 'foreign_field_attribute' set to None

## Actual behavior
An exception is raised while trying to retrieve an instance with 'foreign_field' set to Null:
'NoneType' object has no attribute 'foreign_field'

The above works as expceted in version 3.6.2, but in version 3.7.3 (I do not know where in between the behavior is changed), raises an AttributeError. Seems like a change in fields.py / get_attribute method is causing this.

**Comment by xordoquy** [comment]
Looks like a duplicate of either #3451 and/or #5635

**Comment by ozgurakcali** [comment]
@xordoquy I previously accidentally posted the issue prematurely, could you check again? I am not sure if this is a duplicate of either of the issues you pointed.

**Comment by rpkilby** [comment]
Per the updated description, this looks like a separate case from the referenced issues. In general, I'm throwing this on the stack of issues related to `get_attribute()`

**Comment by tomchristie** [comment]
Not necessarily sure that this is something we'd want to support. I guess it depends if supporting `ancestor=None` -> `serialize as None` would break any existing test cases or not.

**Comment by gthieleb** [comment]
Does this mean that support for source nested nullable foreign keys will be dropped with 3.7? @tomchristie What would be your suggested workaround?

**Comment by carltongibson** [comment]
This should be addressed by #5849. (The exact example may require setting `default=None`.)

**Comment by osumoclement** [comment]
I am encountering a problem with this. In my case, I am trying to use `PrimaryKeyRelatedField` which doesn't work. IntegerField does work, but I really want to use PrimaryKeyRelatedField` for consistency with the rest of my code:

#this doesn't work
`foreign_field_attribute = serializers.PrimaryKeyRelatedField(source='foreign_field.my_attribute', allow_null=True, required=False)`

#this works
`foreign_field_attribute = serializers.IntegerField(source='foreign_field.my_attribute', allow_null=True, required=False)`
